### PR TITLE
Make `add_files` to support `snapshot_properties` argument

### PR DESCRIPTION
**Pull request by enkidulan**
Issue https://github.com/apache/iceberg-python/issues/694

I'm not sure where to add tests. Is `tests/catalog/test_glue.py` the right place?

### CODE REVIEW

**Comment by syun64** [comment]
@enkidulan - thank you for working on this. I think https://github.com/apache/iceberg-python/blob/main/tests/integration/test_add_files.py is the right place for these tests

File: tests/integration/test_add_files.py [hunk 1]
```diff
@@ -122,8 +123,13 @@ def _create_table(
     return tbl
 
 
+@pytest.fixture(name="format_version", params=[pytest.param(1, id="format_version=1"), pytest.param(2, id="format_version=2")])
+def format_version_fixure(request: pytest.FixtureRequest) -> Iterator[int]:
```
Comment (enkidulan) [inline tests/integration/test_add_files.py:127]: Making params more human-friendly: 
```
test_add_files_to_unpartitioned_table[format_version=1] PASSED
test_add_files_to_unpartitioned_table[format_version=2] PASSED
test_add_files_to_unpartitioned_table_raises_file_not_found[format_version=1] PASSED
test_add_files_to_unpartitioned_table_raises_file_not_found[format_version=2] PASSED
test_add_files_to_unpartitioned_table_raises_has_field_ids[format_version=1] PASSED
test_add_files_to_unpartitioned_table_raises_has_field_ids[format_version=2] PASSED
test_add_files_to_unpartitioned_table_with_schema_updates[format_version=1] PASSED
test_add_files_to_unpartitioned_table_with_schema_updates[format_version=2] PASSED
test_add_files_to_partitioned_table[format_version=1] PASSED
test_add_files_to_partitioned_table[format_version=2] PASSED
test_add_files_to_bucket_partitioned_table_fails[format_version=1] PASSED
test_add_files_to_bucket_partitioned_table_fails[format_version=2] PASSED
test_add_files_to_partitioned_table_fails_with_lower_and_upper_mismatch[format_version=1] PASSED
test_add_files_to_partitioned_table_fails_with_lower_and_upper_mismatch[format_version=2] PASSED
test_add_files_snapshot_properties[format_version=1] PASSED
test_add_files_snapshot_properties[format_version=2] PASSED
```

**Comment by enkidulan** [comment]
Thanks @syun64. I've added the tests

**Review by syun64 — APPROVED** [review]
LGTM. Thank you for adding this in!

**Review by kevinjqliu — APPROVED** [review]
lgtm! love the `format_version` change

**Review by HonahX — APPROVED** [review]
It looks great! Thanks @enkidulan.

**Comment by enkidulan** [comment]
I pushed a small update to make Python 3.8 tests pass. I had to replace `collections.abc.Iterator` with `typing.Iterator`, as the former only supports subscripting starting with Python 3.9 - https://docs.python.org/3.11/library/typing.html#typing.Iterator.
Sorry that I haven't done that from the start. For some reason, CI only now notified me about the build failing.

**Comment by HonahX** [comment]
@enkidulan Thanks for fixing the test and the great work! Thanks @syun64 and @kevinjqliu for reviewing. Merging!

**Pull request merged**